sql/core/src/main/scala/org/apache/spark/sql/parquet/ParquetTypes.scala

```diff
@@ -404,7 +405,16 @@ private[parquet] object ParquetTypesConverter extends Logging {
     }
   }
 
+  def checkSpecialCharacters(schema: Seq[Attribute]) = {
+    // ,;{}()\n\t= and space character are special characters in Parquet schema
+    if (schema.exists(_.name.matches(".*[ ,;{}()\n\t=].*"))) {
+      sys.error("""Attribute name can not contain any character among " ,;{}()\n\t=".
+                   | Use Alias to rename attribute name""".stripMargin)
```

Comment:
Let's include the attribute name in the error message to make it more clear, and reformat the code a bit:

```scala
  private def checkSpecialCharacter(schema: Seq[Attribute]) = {
    schema.map(_.name).foreach { name =>
      if (name.matches(".*[ ,;{}()\n\t=].*")) {
        sys.error(
          s"""Attribute name "$name" contains invalid character(s) among " ,;{}()\n\t=".
             |Please use alias to rename it.
           """.stripMargin.split("\n").mkString)
      }
    }
  }
```